docs/site/Inside-Loopback-Application.md

```diff
@@ -0,0 +1,173 @@
+
+[Models](https://loopback.io/doc/en/lb4/Model.html#definition-of-a-model) define the structure of domain objects. [Model Relations](https://loopback.io/doc/en/lb4/Relations.html) help in defining entity composition and cardinalities like one-to-many or one-to-one.
+
+> Models can also be generated from [existing tables](https://loopback.io/doc/en/lb4/Discovering-models.html) in relational databases.
+
+[User](https://github.com/strongloop/loopback4-example-shopping/blob/master/packages/shopping/src/models/user.model.ts) and [Order](https://github.com/strongloop/loopback4-example-shopping/blob/master/packages/shopping/src/models/order.model.ts) are domain objects in the shopping cart use case:
+  - They have the same structure and relations when posted by users from the front end as well as when persisted towards a backend datastore.
+  - A user can have multiple orders, and hence `User` model has a [has-many relation](https://loopback.io/doc/en/lb4/HasMany-relation.html) (ie., one-to-many) with `Order` model. Hence the `orders` property in `User` model is annotated with the `@hasMany(() => Order)` decorator.
+  - An `Order` is composed with a list of products added in the shopping cart and submitted towards the order. Hence the `products` property in `Order` model is annotated as `@property.array()` to indicate that it is a list property.
+  - An `Order` can be made only by one `User`. So, the `userId` property in `Order` model is annotated with `@belongsTo()` decorator, to indicate `Order` has a one-to-one relation with `User`.
+
+## Entity layer and Database connectivity
+
+Applications displaying a `user` profile might need a list of associated recent `orders` made by the user. The same also applies for an order page, some associated user data could be needed. Querying for associated data for demands from front end as well as for logical and persistence reasons is a common requirement. Having a separate `Entity` layer helps in model driven CRUD operations.
+
+> [Repositories](https://loopback.io/doc/en/lb4/Repositories.html) in LoopBack represent the `Entity` layer for querying and persisting a domain model. They also connect with other repositories to resolve `Entity relations`.
+
+From the exampple,
+  - To query and persist `User` model data, the example uses a [UserRepository](https://github.com/strongloop/loopback4-example-shopping/blob/master/packages/shopping/src/repositories/user.repository.ts).
+  - For the `Order` model, [OrderRepository](https://github.com/strongloop/loopback4-example-shopping/blob/master/packages/shopping/src/repositories/order.repository.ts) is defined.
+  - The `UserRepository` uses the entity composition defined in the `User` model to create a has-many relation with the `OrderRepository` 
+
+We need appropriate drivers to connect with the back end datasources.
```

Comment:
should it be `backend` instead of `back end`?

Reply:
agreed

Reply:
![image](https://user-images.githubusercontent.com/6864736/67293620-ab7d9180-f4b2-11e9-8967-22e9db63c1ba.png)

Reply:
backend is valid

Reply:
thanks for the dictionary reference @emonddr :)